apps/desktop/src-tauri/src/inference/server.rs

```diff
@@ -58,24 +58,55 @@ async fn post_completions(payload: Json<CompletionRequest>) -> impl Responder {
 
   let (token_sender, receiver) = flume::unbounded::<Bytes>();
 
-  HttpResponse::Ok()
-    .append_header(("Content-Type", "text/event-stream"))
-    .append_header(("Cache-Control", "no-cache"))
-    .keep_alive()
-    .streaming({
-      let abort_flag = Arc::new(RwLock::new(false));
-
-      AbortStream::new(
-        receiver,
-        abort_flag.clone(),
-        start(InferenceThreadRequest {
-          model_guard: model_guard.clone(),
-          abort_flag: abort_flag.clone(),
-          token_sender,
-          completion_request: payload.0,
-        }),
-      )
-    })
+  if let Some(true) = payload.stream {
+    HttpResponse::Ok()
+      .append_header(("Content-Type", "text/event-stream"))
+      .append_header(("Cache-Control", "no-cache"))
+      .keep_alive()
+      .streaming({
+        let abort_flag = Arc::new(RwLock::new(false));
+        let str_buffer = Arc::new(Mutex::new(String::new()));
+
+        AbortStream::new(
+          receiver,
+          abort_flag.clone(),
+          start(InferenceThreadRequest {
+            model_guard: model_guard.clone(),
+            abort_flag: abort_flag.clone(),
+            token_sender,
+            completion_request: payload.0,
+            nonstream_completion_tokens: str_buffer.clone(),
+            stream: true,
+            tx: None,
+          }),
```

Comment:
I have this idea which I think would make this nicer - we can create the `InferenceThreadRequest` before the isStream check actually, since it's non-blocking state. We can then do

```rust
let request = InferenceThreadRequest {
            model_guard: model_guard.clone(),
            abort_flag: abort_flag.clone(),
            token_sender,
            completion_request: payload.0,
            nonstream_completion_tokens: str_buffer.clone(),          
}

if request.isStream() {} else {} 
```

And the `.isStream` is a trait public method we expose via InferenceThreadRequest, which basically return `completion_request.stream`

Reply:
I really like your attention to detail and design thinking! I will try to implement this one, I agree, it is indeed cleaner.